### PR TITLE
Move StringContext test to junit

### PR DESCRIPTION
**Pull request by som-snytt**

### CODE REVIEW

File: test/files/neg/t5856.scala [hunk 1]
```diff
@@ -8,4 +8,12 @@ object Test {
   val s7 = s"$s1 $null $super"
   val s8 = s"$super"
   val s9 = s"$"
+  val sA = s"$this"
```
Comment (NthPortal) [inline test/files/neg/t5856.scala:11]: this is a `neg` test - shouldn't `s"$this"` compile fine? (it appears to in the junit test)

Comment (lrytz) [reply test/files/neg/t5856.scala:11]: this is still useful in a neg test because it tests that no warning / error is issued

**Review by lrytz — APPROVED** [review]

File: test/files/neg/t5856.scala (hunk 1 above)
Comment (som-snytt) [reply test/files/neg/t5856.scala:11]: Sorry Nth, I almost added a comment at that line that the rest compiles. Then I didn't. Maybe I fell asleep? Maybe my spouse arrived home late from work at that moment? This delta was in an old PR that was sadly rejected.

**Comment by som-snytt** [comment]
Amazed at the maintainers' chutzpah in merging things with red exes. I could never do that.

File: test/files/neg/t5856.scala (hunk 1 above)
Comment (som-snytt) [reply test/files/neg/t5856.scala:11]: oh except I changed assert to assertEquals, just because.

**Pull request merged**